### PR TITLE
Extract minimum supported PHP version from plugin file for GH actions

### LINKED ISSUE #6815: Build zip workflow - use minimum PHP required to allow running on older versions of the plugin
### Description
In some cases (backport fixes), we may want to run the [Build zip file & Run smoke tests](https://github.com/Automattic/woocommerce-payments/actions/workflows/build-zip-and-run-smoke-tests.yml) on branches that are based on older versions.

Those older versions could have a lower PHP requirement compared to our current production version.

Today, we require minimum PHP 7.3 but last year, we required minimum PHP 7.1.

I can think of two solutions:
- add an optional PHP version input
- have the workflow figure out the minimum PHP requirement from the branch it runs on (to be parsed [here](https://github.com/Automattic/woocommerce-payments/blob/3471a02a18ca09479f6861058bdc1b4a99642e82/woocommerce-payments.php#L14))

### Acceptance criteria

* We can run the workflow on branches that require lower PHP version compared to the latest version

### Additional context
p1689764318051319/1689747102.965849-slack-C05AW31EKFF

### PR DESCRIPTION
**Pull request by deepakpathania**
Fixes #6815 

#### Changes proposed in this Pull Request

- Create a new action to setup PHP that extracts the version from plugin file.
- Update usage to use new composite action.

#### Testing instructions

* Workflows on this branch should work as expected - [sample run](https://github.com/Automattic/woocommerce-payments/actions/runs/6158485079). 

-------------------

- [ ] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [ ] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `7179` or branch name `dev/update-workflows` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=dev/update-workflows) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 16e3400313b435534053789a8f086e2b3daf2c0f
- Build time: 2023-09-12 11:55:46 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: .github/actions/setup-php/action.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+name: "Setup PHP"
```
Comment (jbordonado) [inline .github/actions/setup-php/action.yml]:
```suggestion
name: "Set up PHP"
```

File: changelog/dev-update-workflows [hunk 2]
```diff
@@ -0,0 +1,4 @@
+Significance: patch
+Type: dev
+
+Update GH workflows to use PHP version from plugin file.
```
Comment (jbordonado) [inline changelog/dev-update-workflows]: It should be a non-significant changelog, so it doesn't appear on the next release's changelog

**Review by jbordonado — APPROVED** [review]
Pre-approving with 2 minor comments

**Pull request merged**